pandas/core/window.py

```diff
@@ -707,63 +813,20 @@ def _pop_args(win_type, arg_names, kwargs):
             # GH #15662. `False` makes symmetric window, rather than periodic.
             return sig.get_window(win_type, window, False).astype(float)
 
-    def _apply_window(self, mean=True, **kwargs):
-        """
-        Applies a moving window of type ``window_type`` on the data.
+    def _get_roll(self, func, check_minp, *args, **kwargs):
 
-        Parameters
-        ----------
-        mean : bool, default True
-            If True computes weighted mean, else weighted sum
-
-        Returns
-        -------
-        y : same type as input argument
-
-        """
-        window = self._prep_window(**kwargs)
-        center = self.center
-
-        blocks, obj, index = self._create_blocks()
-        block_list = list(blocks)
-
-        results = []
-        exclude = []
-        for i, b in enumerate(blocks):
-            try:
-                values = self._prep_values(b.values)
-
-            except (TypeError, NotImplementedError):
-                if isinstance(obj, ABCDataFrame):
-                    exclude.extend(b.columns)
-                    del block_list[i]
-                    continue
-                else:
-                    raise DataError("No numeric types to aggregate")
-
-            if values.size == 0:
-                results.append(values.copy())
-                continue
-
-            offset = _offset(window, center)
-            additional_nans = np.array([np.NaN] * offset)
-
-            def f(arg, *args, **kwargs):
-                minp = _use_window(self.min_periods, len(window))
-                return libwindow.roll_window(
-                    np.concatenate((arg, additional_nans)) if center else arg,
-                    window,
-                    minp,
-                    avg=mean,
-                )
-
-            result = np.apply_along_axis(f, self.axis, values)
+        cfunc = getattr(libwindow, func, None)
```

Comment:
shouldn’t this be checked at a higher level? (in _apply)

Reply:
Could be. Will change function name as wrap_roll then